UnityProject/Assets/Scripts/Objects/Other/MouseTrap.cs

```diff
@@ -41,40 +58,99 @@ private void HurtHand(LivingHealthMasterBase health)
 		/// Things to trigger for attached items if found + hurting peoples hands.
 		/// </summary>
 		/// <param name="health"></param>
-		public void TriggerTrapFromContainer(LivingHealthMasterBase health)
+		public void TriggerTrap(LivingHealthMasterBase health = null)
 		{
 			if(health != null) HurtHand(health);
-			//TODO : Add the ability to attach items to mousetraps and trigger them.
+			var slot = trapContent.GetTopOccupiedIndexedSlot();
+			if(slot == null) return;
+			if (slot.ItemObject.TryGetComponent<ITrapComponent>(out var component))
+			{
+				if(Inventory.ServerDrop(slot)) trapInSnare = false;
+				component.TriggerTrap();
+			}
+			UpdateTrapVisual();
+			isArmed = false;
+		}
+
+		private bool HasTrapTrait(GameObject item)
+		{
+			if (item.Item() != null)
+			{
+				if (item.Item().HasTrait(trapTrait)) return true;
+			}
+			return false;
 		}
 
+		private void UpdateTrapVisual()
+		{
+			if (trapContent.GetNextFreeIndexedSlot() != null)
+			{
+				trapPreview.Empty();
+				return;
+			}
+			//We assume that there will be only one item on each mouse trap (for now)
+			var slot = trapContent.GetTopOccupiedIndexedSlot();
+			var sprite = slot.Item.gameObject.GetComponentInChildren<SpriteHandler>();
+			if (sprite.GetCurrentSpriteSO() == null)
+			{
+				trapPreview.SetSprite(sprite.CurrentSprite);
+				return;
+			}
+			trapPreview.SetSpriteSO(sprite.GetCurrentSpriteSO());
+		}
 
 		public override void OnStep(GameObject eventData)
 		{
 			if (IsArmed == false) return;
+			if(trapInSnare) TriggerTrap();
+			//a mouse trap must kill mice, duh
+			//TODO : IEnterable is designed for players only so mice can't trigger this :(
```

Comment:
Yeah i've fixed this (added an new interface for objects) in the stuff im working on